src/e2ir.c

```diff
@@ -834,32 +834,18 @@ elem *setArray(elem *eptr, elem *edim, Type *tb, elem *evalue, IRState *irs, int
     return e;
 }
 
-struct StringTab
-{
-    Module *m;          // module we're generating code for
-    Symbol *si;
-    void *string;
-    size_t sz;
-    size_t len;
-};
-
-#define STSIZE 16
-StringTab stringTab[STSIZE];
-size_t stidx;
 
-static Symbol *assertexp_sfilename = NULL;
-static const char *assertexp_name = NULL;
-static Module *assertexp_mn = NULL;
+StringTable *stringTab;
 
+/********************************
+ * Reset stringTab[] between object files being emitted, because the symbols are local.
+ */
 void clearStringTab()
 {
     //printf("clearStringTab()\n");
-    memset(stringTab, 0, sizeof(stringTab));
-    stidx = 0;
-
-    assertexp_sfilename = NULL;
-    assertexp_name = NULL;
-    assertexp_mn = NULL;
+    delete stringTab;
+    stringTab = new StringTable;
```

Comment:
This is inefficient - can't you reuse the same string table instead of deallocating and reallocating?